File: code/functions/callbacks.erl
% Callbacks in Erlang
-module(example).
-export([start/0]).

callback(Value) -&gt;
    io:format("Callback called with: ~p~n", [Value]).

process_with_callback(Callback, Data) -&gt;
    Callback(Data).

start() -&gt;
    process_with_callback(fun callback/1, 42).

File: code/functions/callbacks.java
// Callbacks in Java
interface Callback {
    void call(int value);
}

public class Example {
    public static void processWithCallback(Callback cb, int data) {
        cb.call(data);
    }
    
    public static void main(String[] args) {
        processWithCallback(new Callback() {
            public void call(int value) {
                System.out.println("Callback called with: " + value);
            }
        }, 42);
    }
}